src/components/OSCALMetadata.js

```diff
@@ -38,17 +53,215 @@ const OSCALMetadataPartiesCard = styled(Paper)(({ theme }) => ({
+    emptyMessage = "No information provided"
+  ) => {
+    if (!list?.length) {
+      return <Typography> {emptyMessage} </Typography>;
+    }
+
+    return list.map((item) => (
+      <ListItem
+        key={partyInfoType === "email" ? item : `${item?.type}--${item?.name}`}
+      >
+        {TYPE_MAPPING(item)[partyInfoType]}
+      </ListItem>
+    ));
+  };
+
+  return (
+    <CardActions>
+      <Button size="small" onClick={handleOpen}>
+        Contact
+      </Button>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        scroll="paper"
+        aria-labelledby="scroll-dialog-title"
+        aria-describedby="scroll-dialog-description"
+        maxWidth="md"
+        fullWidth
+      >
+        <DialogTitle id="scroll-dialog-title">{props.party.name}</DialogTitle>
+        <DialogContent>
+          <Grid container spacing={1}>
+            <Grid item xs={4}>
+              <Typography variant="h6" component="h3">
+                <MapIcon />
+                Addresses
+              </Typography>
+              <List>
+                {getPartyInfoList(
+                  props.party.addresses,
+                  PartyInfoTypes.address,
+                  "No address information provided"
+                )}
+              </List>
+            </Grid>
+
+            <Grid item xs={4}>
+              <Typography variant="h6" component="h3">
+                <PhoneIcon />
+                Phone
+              </Typography>
+              <List>
+                {getPartyInfoList(
+                  props.party["telephone-numbers"],
+                  PartyInfoTypes.telephone,
+                  "No telephone information provided"
+                )}
+              </List>
+            </Grid>
+            <Grid item xs={4}>
+              <Typography variant="h6" component="h3">
+                <AlternateEmailIcon />
+                Email
+              </Typography>
+              <List>
+                {getPartyInfoList(
+                  props.party["email-addresses"],
+                  PartyInfoTypes.email,
+                  "No email information provided"
+                )}
+              </List>
+            </Grid>
+          </Grid>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleClose} color="primary">
+            Close
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </CardActions>
+  );
+}
+
+export function OSCALMetadataParty(props) {
+  return (
+    <Card>
+      <CardContent>
+        <Grid container spacing={2}>
+          <Grid item>
+            <Avatar>
+              {props.party.type === "organization" ? <GroupIcon /> : null}
+            </Avatar>
+          </Grid>
+          <Grid item>
+            <Typography>
+              {props.party.name}
+              {props.partyRolesText}
+            </Typography>
+          </Grid>
+        </Grid>
+      </CardContent>
+      <OSCALMetadataPartyDialog party={props.party} />
```

Comment:
```suggestion
            </Typography>
            <OSCALMetadataPartyDialog party={props.party} />
          </Grid>
        </Grid>
      </CardContent>
```

This addresses my previous suggestion.